### DISCUSSION TITLE
Filepicker behavior differences between iOS and Android

### DISCUSSION THREAD

**Discussion by cwiggins999** [capawesome-team/capacitor-plugins#170] [Q&A]
The Filepicker is working well for us on both mobile platforms. But there seem to be some inherent differences in how files are managed between iOS and Android that impact our use. (I'm much more familiar with iOS than Android as far as general feature sets like file management.) Our interest is primarily in allowing media files - audio/video/images - to be chosen for upload.

On iOS Filepicker takes us to a "Files" app view. So I can create a folder and place images there through the files app, then the file picker can see only those files. Specifically, you cannot see the entire camera roll, which is good!

Android seems to open up the all the media on the device for me - at least all that's included in the photo gallery and other media sources - music, videos, etc. 

We clearly don't want the user to be able to upload their entire media library. We can restrict to a certain number of files that we upload, but it would be better if there were only a certain set of folders available, so that users would need to copy their desired media files to that location, which would put controls of sorts on it.

I'm probably talking about a bigger issue of how Android manages files - just trying to identify the right questions and path forward.

Thanks!

Chuck Wiggins

**Comment by robingenz** [comment]
> We can restrict to a certain number of files that we upload, but it would be better if there were only a certain set of folders available, so that users would need to copy their desired media files to that location, which would put controls of sorts on it.

I don't think that's possible. At least I don't know of any API that allows this.

**Reply by cwiggins999** [reply]
On Android, In theory it seems it would be fairly easy for a user to "select all" and have a thousand or more filenames passed into the plugin file array. Have you seen any such things, or is there something that would prevent that? I saw some discussions of limiting the number via a plugin option, but didn't sound like that's possible on Android.

**Reply by robingenz** [reply]
The question is why should a user do this. I recommend checking the number of files selected and displaying a message to the user if they have selected too many.

> I saw some discussions of limiting the number via a plugin option, but didn't sound like that's possible on Android.

It's only possible using the new [Photo picker](https://developer.android.com/training/data-storage/shared/photopicker). There is already a feature request for it: #107

**Comment by cwiggins999** [comment]
The user shouldn't do it, but I think it's possible anytime there is a "select all" option on a file/media selection view. I never underestimate the potential for people to do things that don't make sense :-)
